libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -448,6 +449,33 @@ void AP_DDS_Client::on_topic(uxrSession* uxr_session, uxrObjectId object_id, uin
             GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received sensor_msgs/Joy: %f, %f, %f, %f",
                           rx_joy_topic.axes[0], rx_joy_topic.axes[1], rx_joy_topic.axes[2], rx_joy_topic.axes[3]);
             // TODO implement joystick RC control to AP
+            const uint32_t tnow = AP_HAL::millis();
+            
+            for (uint8_t i=0; i<rx_joy_topic.axes_size; i++) {
```

Comment:
The loop condition has changed from the original example. It was:

```c++
        for (uint8_t i=0; i<MIN(8U, rx_joy_topic.axes_size); i++) {
```

This is to match the implementation in `GCS_Common.cpp` : `GCS_MAVLINK::handle_rc_channels_override`. The MIN is needed because we also need to check the size of axes_size.